### PR TITLE
Deprecating vector search APIs and updating README accordingly

### PR DESCRIPTION
**Pull request by cjnolet**
I opted to deprecate just the necessary pieces, such as the `index` classes instead of deprecating every single function.

### CODE REVIEW

File: cpp/include/raft/neighbors/epsilon_neighborhood.cuh [hunk 1]
```diff
@@ -98,12 +98,13 @@ void epsUnexpL2SqNeighborhood(bool* adj,
  *                    squared as we compute L2-squared distance in this method)
  */
 template <typename value_t, typename idx_t, typename matrix_idx_t>
-void eps_neighbors_l2sq(raft::resources const& handle,
-                        raft::device_matrix_view<const value_t, matrix_idx_t, row_major> x,
-                        raft::device_matrix_view<const value_t, matrix_idx_t, row_major> y,
-                        raft::device_matrix_view<bool, matrix_idx_t, row_major> adj,
-                        raft::device_vector_view<idx_t, matrix_idx_t> vd,
-                        value_t eps)
+[[deprecated("Use cuVS instead")]] void eps_neighbors_l2sq(
```
Comment (benfred) [inline cpp/include/raft/neighbors/epsilon_neighborhood.cuh:101]: Has this function been moved to cuvs?

**Review by benfred — APPROVED** [review]

File: cpp/include/raft/neighbors/epsilon_neighborhood.cuh (hunk 1 above)
Comment (cjnolet) [reply cpp/include/raft/neighbors/epsilon_neighborhood.cuh]: Nope, it hasn't :-(. I thought it had but I should have verified. Thank you for pointing this out! Not a high priority so I just removed it from the deprecations for now. I'll add this to our tracker issue after code freeze begins.

File: docs/source/raft_ann_benchmarks.md [hunk 2]
```diff
@@ -279,15 +283,15 @@ python -m raft_ann_bench.plot --dataset deep-image-96-inner
 
 Configuration files already exist for the following list of the million-scale datasets. Please refer to [ann-benchmarks datasets](https://github.com/erikbern/ann-benchmarks/#data-sets) for more information, including actual train and sizes. These all work out-of-the-box with the `--dataset` argument. Other million-scale datasets from `ann-benchmarks.com` will work, but will require a json configuration file to be created in `$CONDA_PREFIX/lib/python3.xx/site-packages/raft_ann_bench/run/conf`, or you can specify the `--configuration` option to use a specific file.
 
-| Dataset Name | Train Rows | Columns | Test Rows      | Distance   | 
+**| Dataset Name | Train Rows | Columns | Test Rows      | Distance   |
```
Comment (benfred) [inline docs/source/raft_ann_benchmarks.md:286]: Adding the `**` around this section makes this render a bit weird in GH ui at least - was this intentional?

File: docs/source/raft_ann_benchmarks.md [hunk 3]
```diff
@@ -287,7 +291,7 @@ Configuration files already exist for the following list of the million-scale da
 | `glove-100-angular` | 1.1M | 100 | 10K | Angular |
 | `mnist-784-euclidean` | 60K | 784 | 10K | Euclidean |
 | `nytimes-256-angular` | 290K | 256 | 10K | Angular |
-| `sift-128-euclidean` | 1M | 128 | 10K | Euclidean|
+| `sift-128-euclidean` | 1M | 128 | 10K | Euclidean|**
```
Comment (benfred) [inline docs/source/raft_ann_benchmarks.md:294]: This change here should also be reverted

Comment (cjnolet) [reply docs/source/raft_ann_benchmarks.md:294]: lol, yep it certainly should

**Pull request merged**